integration/container: Deflake TestWaitBlocked and TestWaitRestartedContainer

vvoland opened the pull request:
- fix: https://github.com/moby/moby/issues/49595
- fix: https://github.com/moby/moby/issues/48983

Remove Parallel execution and increase stop timeout seems to help.

vvoland committed:
integration/container: Increase stop timeout for TestWaitRestartedContainer

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

robmry commented:
Does the `t.Parallel` break it because some of the container-stop is serialised in the daemon, so the stops end up queued anyway, then 2s isn't enough for the last in the queue? (Just wondering increasing the 2s timeout works as an alternative, because testing stops running in parallel might uncover something interesting?)

vvoland commented:
Nope, the 10s timeout is hit on the daemon side so it's not about the 2s wait timeout.

I don't have a good explanation yet why removing Parallel works. Or maybe it actually doesn't and I wasn't able to hit the jackpot 🙈 

I'm still willing to investigate this more, but would like to first make sure that removing Parallel is the key here.

vvoland commented:
Actually, let me try to test one more change on this one.

vvoland committed:
integration/container: Remove Parallel from TestWait*

It seems to help with the flakiness in the CI.
However, I can't reproduce the flakiness locally.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland commented:
I can't reproduce the flakiness locally so I think it's related to the system load of the CI machine? 🤷🏻

robmry reviewed (approved):
Thank you for investigating ... is the conclusion that parallel stops actually are flaky in the daemon, so it's not "just" a test problem? (Do we need an issue to track that?)

Either way, we can't have flaky tests - so this change LGTM.

vvoland commented:
No concrete idea yet, for some reason the container just doesn't exit after the daemon sends the SIGTERM.
Perhaps the `trap 'exit 0' TERM` doesn't have the chance to execute before `ContainerStop` is called? 🤔 But I have no explanation why removing the Parallel would help that.

I'll open a ticket for it and keep an eye on this test.
